### ISSUE TITLE
java.lang.IllegalStateException while running the generated test suite

### ISSUE THREAD

**Issue by jose** [EMResearch/EMB#24]
Hi @arcuri82,

I have successfully executed [EvoMaster](https://github.com/EMResearch/EvoMaster) on the [features-service](https://github.com/EMResearch/EMB/tree/master/jdk_8_maven/cs/rest/original/features-service) project and also managed to successfully compile the generated test suite.  Question now is, how do I execute the generated test suite, [EM__0_Test.java](https://github.com/EMResearch/EMB/files/7786056/EM__0_Test.txt) ?

I have tried the following command:

```bash
java -cp .:../../features-service-sut.jar:../../features-service-evomaster-runner.jar:$HOME/.m2/repository/junit/junit/4.11/junit-4.11.jar:$HOME/.m2/repository/org/hamcrest/hamcrest-all/1.3/hamcrest-all-1.3.jar:$HOME/.m2/repository/io/rest-assured/rest-assured/4.3.0/rest-assured-4.3.0.jar:$HOME/.m2/repository/io/rest-assured/json-path/4.3.0/json-path-4.3.0.jar \
    org.junit.runner.JUnitCore \
    EM__0_Test
```

but does not seem to be working

```bash
JUnit version 4.11
Going to start the SUT
Going to stop the SUT
E
Time: 0.016
There was 1 failure:
1) EM__0_Test
java.lang.IllegalStateException: Invalid jar path does not end with '.jar': ../core/target
        at org.evomaster.client.java.controller.ExternalSutController.validateJarPath(ExternalSutController.java:387)
        at org.evomaster.client.java.controller.ExternalSutController.startSut(ExternalSutController.java:147)
        at EM__0_Test.initClass(EM__0_Test.java:62)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:47)
        at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
        at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:44)
        at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:24)
        at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
        at org.junit.runners.ParentRunner.run(ParentRunner.java:309)
        at org.junit.runners.Suite.runChild(Suite.java:127)
        at org.junit.runners.Suite.runChild(Suite.java:26)
        at org.junit.runners.ParentRunner$3.run(ParentRunner.java:238)
        at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:63)
        at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:236)
        at org.junit.runners.ParentRunner.access$000(ParentRunner.java:53)
        at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:229)
        at org.junit.runners.ParentRunner.run(ParentRunner.java:309)
        at org.junit.runner.JUnitCore.run(JUnitCore.java:160)
        at org.junit.runner.JUnitCore.run(JUnitCore.java:138)
        at org.junit.runner.JUnitCore.run(JUnitCore.java:117)
        at org.junit.runner.JUnitCore.runMain(JUnitCore.java:96)
        at org.junit.runner.JUnitCore.runMainAndExit(JUnitCore.java:47)
        at org.junit.runner.JUnitCore.main(JUnitCore.java:40)

FAILURES!!!
Tests run: 0,  Failures: 1
```

Any clue?

#### Setup

- Linux
- JDK 1.8.0_292
- [EMB 1.3.1-SNAPSHOT](https://github.com/EMResearch/EMB/commit/5f81fa65daf463cb7aec43a7b22ccafdf653479b)
- EvoMaster 1.3.1-SNAPSHOT (commit [9810a07](https://github.com/EMResearch/EvoMaster/commit/9810a07ee456357a8def2d61612741f1a51c26a2))

**Comment by arcuri82** [comment]
Hi,
 that's a known "limitation", as you are using the `ExternalSutController`, instead of Embedded one. for the former, we usually do not run the generated tests for those SUTs in EMB, so we never fixed the file paths to the JAR files containing the SUTs. 
But we should...
Anyway, meanwhile, if you are doing anything with these generated tests (eg measuring code coverage or mutation score), you should rather use the Embedded controllers, as External does run the SUT in a separated process.

**Comment by jose** [comment]
Thanks for the feedback.  How should I use the Embedded one instead of the `ExternalSutController`?  Is it just a matter of replacing the following line:
```java
private static final SutHandler controller = new em.external.org.javiermf.features.ExternalEvoMasterController();
```
with
```java
private static final SutHandler controller = new em.embedded.org.javiermf.features.EmbeddedEvoMasterController();
```
in the generated test suite and adding the `jdk_8_maven/em/embedded/rest/features-service/target/evomaster-benchmark-em-embedded-rest-features-service-1.3.1-SNAPSHOT.jar` (and any other dependency) to the classpath?

Or do I need to somehow generate tests in a different way?  By the way, here are the commands I used to generate the [test suite](https://github.com/EMResearch/EMB/files/7786056/EM__0_Test.txt):

```bash
/bin/java -Xms1G -Xmx4G \
  -Dem.muteSUT=true \
  -Devomaster.instrumentation.jar.path=evomaster-agent.jar \
  -jar features-service-evomaster-runner.jar \
  10210 10211 \
  features-service-sut.jar \
  1200 \
  /bin/java > log_sut_features-service_10210.txt 2>&1 &

CONTROLLER_PID=$!
sleep 20

/bin/java -Xms2G -Xmx4G \
  -jar evomaster.jar \
  --testSuiteFileName=EM__0_Test \
  --stoppingCriterion=FITNESS_EVALUATIONS \
  --maxActionEvaluations=10000 \
  --statisticsColumnId=features-service \
  --seed=0 \
  --sutControllerPort=10210 \
  --outputFolder=tests/ \
  --statisticsFile=statistics_features-service_0.csv \
  --snapshotInterval=5 \
  --snapshotStatisticsFile=snapshot_features-service_0.csv \
  --appendToStatisticsFile=true \
  --writeStatistics=true \
  --showProgress=false \
  --testSuiteSplitType=NONE >> log_em_features-service_10210.txt 2>&1

kill "$CONTROLLER_PID"
```

**Comment by arcuri82** [comment]
Hi,
 you have 2 options:

1) change the generated tests manually, like you wrote, by changing `em.external.org.javiermf.features.ExternalEvoMasterController` into `em.embedded.org.javiermf.features.EmbeddedEvoMasterController`

2) use `em.embedded.org.javiermf.features.EmbeddedEvoMasterController` directly when starting the controller, instead of External, when generating tests. But this will work from an IDE (IDEA or Eclipse) but not the command line (due to classloading)

**Comment by arcuri82** [comment]
Hi. this issue should be now fixed in the new 1.4.0 release
